alica_tests/include/test_alica.h

```diff
@@ -25,17 +26,10 @@
 namespace alica
 {
 
-// Don't want to expose engine to app developers
-struct AlicaTestEngineGetter
-{
-    static alica::AlicaEngine* getEngine(alica::AlicaContext* ac) { return ac->_engine.get(); }
-};
```

Comment:
That's a lot of change that we don't need. alica tests should be allowed to get to the engine and verify finer details. If you add everything to the TestContext it will become bloated and confuse application developers. TestContext is supposed to be used only by alica application tests and not alica tests yhemselves. You can revert all changes to the tests by keeping the engine getter a friend of AlicaContext as before, along with TestContext.